src/hooks/useAssetLogoSource.ts

```diff
@@ -0,0 +1,77 @@
+import TokenLogoLookupTable, { BAD_SRCS } from 'constants/TokenLogoLookupTable'
+import { chainIdToNetworkName, getNativeLogoURI } from 'lib/hooks/useCurrencyLogoURIs'
+import uriToHttp from 'lib/utils/uriToHttp'
+import { useCallback, useEffect, useState } from 'react'
+import { isAddress } from 'utils'
+
+// Converts uri's into fetchable urls
+function parseLogoSources(uris: string[]) {
+  const urls: string[] = []
+  uris.forEach((uri) => urls.push(...uriToHttp(uri)))
+  return urls
+}
+
+// Parses uri's, favors non-coingecko images, and improves coingecko logo quality
+function prioritizeLogoSources(uris: string[]) {
+  const parsedUris = uris.map((uri) => uriToHttp(uri)).flat(1)
+  const preferredUris: string[] = []
+  let coingeckoUrl: string | undefined = undefined
```

Comment:
This should be an array, in case there are multiple coingecko URLs. Unless you're trying to consolidate them all, in which case that should be made clear with a comment.